product_harmonized_system_country/models/hs_code_country.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HSCodeCountry(models.Model):
+    _name = "hs.code.country"
+    _description = "H.S. Code Country"
+
+    country_ids = fields.Many2many(
+        string="Countries",
+        comodel_name="res.country",
+    )
+    local_code = fields.Char(
+        required=True,
+    )
+    hs_code = fields.Char(
+        string="H.S. Code",
+        compute="_compute_hs_code",
+        store=True,
+        precompute=True,
+    )
+    hs_code_id = fields.Many2one(
+        string="H.S. Code",
+        comodel_name="hs.code",
+        ondelete="cascade",
+    )
+
+    @api.constrains("country_ids")
+    def _check_country_ids(self):
+        for sel in self:
+            for country in sel.country_ids:
+                if (
+                    len(
+                        sel.hs_code_id.hs_code_country_ids.filtered(
```

Comment:
https://geekflare.com/es/set-operations-in-python/ --> diría que con teoría de conjuntos podría realizarse más rápido. y te podrías evitar el filtered 
(pégale un ojito y me dices :D )

Reply:
I've had a look but I don't get exactly the way you'd like to apply it in here.